backends/ebpf/run-ebpf-test.py

```diff
@@ -26,126 +26,152 @@
+    "-t",
+    "--target",
+    dest="target",
+    default="test",
+    help="Specify the compiler backend target, " "default is test",
+)
+PARSER.add_argument(
+    "-e",
+    "--extern-file",
+    dest="extern",
+    default="",
+    help="Specify path additional file with C extern function definition",
+)
+PARSER.add_argument(
+    "-tf",
+    "--testfile",
+    dest="testfile",
+    help="Provide the path for the stf file for this test. "
+    "If no path is provided, the script will search for an"
+    " stf file in the same folder.",
+)
+PARSER.add_argument(
+    "-ll",
+    "--log_level",
+    dest="log_level",
+    default="WARNING",
+    choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG", "NOTSET"],
+    help="The log level to choose.",
+)
+
 
 def import_from(module, name):
-    """ Try to import a module and class directly instead of the typical
-        Python method. Allows for dynamic imports. """
+    """Try to import a module and class directly instead of the typical
+    Python method. Allows for dynamic imports."""
     module = __import__(module, fromlist=[name])
     return getattr(module, name)
 
 
-class EBPFFactory(object):
-    """ Generator class.
-     Returns a target subclass based on the provided target option."""
+class EBPFFactory:
+    """Generator class.
+    Returns a target subclass based on the provided target option."""
+
     @staticmethod
-    def create(tmpdir, options, template, outputs):
+    def create(tmpdir, options, template):
         target_name = "targets." + options.target + "_target"
         target_class = "Target"
-        report_output(outputs["stdout"], options.verbose,
-                      "Loading target ", target_name)
+        testutils.log.info(f"Loading target {target_name}")
         try:
             EBPFTarget = import_from(target_name, target_class)
         except ImportError as e:
-            report_err(outputs["stderr"], e)
+            testutils.log.error(e)
             return None
-        return EBPFTarget(tmpdir, options, template, outputs)
+        return EBPFTarget(tmpdir, options, template)
 
 
-class Options(object):
+class Options:
     def __init__(self):
-        self.binary = ""                # This program's name
-        self.cleanupTmp = True          # Remove tmp folder?
-        self.compiler = ""              # Path to the P4 compiler binary
-        self.p4Filename = ""            # File that is being compiled
-        self.testfile = ""              # path to stf test file that is used
-        self.verbose = False            # Enable verbose output
-        self.replace = False            # Replace previous outputs
-        self.target = "test"            # The name of the target compiler
-        # Actual location of the test framework
-        self.testdir = os.path.dirname(os.path.realpath(__file__))
-        self.extern = ""                # Path to C file with extern definition
+        self.binary = ""  # This program's name.
```

Comment:
I find the original version easier to read. Why spend time reformatting whitespace?